src/OrchardCore.Modules/OrchardCore.Shortcodes/Services/ShortcodeOptionsTableProvider.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Options;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Shortcodes.Services
+{
+    public class ShortcodeOptionsTableProvider : IShortcodeTableProvider
```

Comment:
This class returns ShortcodeDescriptor instances registered in the IOption<foo>.